lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -107,25 +107,33 @@ def upload_binary(app_id, binary_path)
       #
       # Returns the release_id on a successful release, otherwise returns nil.
       #
-      # Throws an error if the number of polling retries exceeds MAX_POLLING_RETRIES
+      # Throws a UI error if the number of polling retries exceeds MAX_POLLING_RETRIES
+      # or if it was not able to upload the binary
       def upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
         upload_status_response = get_upload_status(app_id, upload_token)
-        if upload_status_response.success?
+        if upload_status_response.success? || upload_status_response.already_uploaded?
           UI.success("This APK/IPA has been uploaded before. Skipping upload step.")
         else
-          UI.message("This APK has not been uploaded before.")
+          UI.message("This APK/IPA has not been uploaded before")
+          UI.message("Uploading the APK/IPA.")
+          unless upload_status_response.in_progress?
+            upload_binary(app_id, binary_path)
+          end
           MAX_POLLING_RETRIES.times do
-            if upload_status_response.success?
+            upload_status_response = get_upload_status(app_id, upload_token)
+            if upload_status_response.success? || upload_status_response.already_uploaded?
               UI.success("Uploaded APK/IPA Successfully!")
               break
             elsif upload_status_response.in_progress?
               sleep(POLLING_INTERVAL_SECONDS)
             else
-              UI.message("Uploading the APK/IPA.")
-              upload_binary(app_id, binary_path)
+              if !upload_status_response.message.nil?
+                UI.crash!("#{ErrorMessage::UPLOAD_APK_ERROR}: #{upload_status_response.message}")
```

Comment:
Consider using `user_error!` since this an expected user error e.g. an incorrectly formatted file.